stdlib/dateBoundaries/dateBoundaries.flux

```diff
@@ -0,0 +1,417 @@
+// import "dateBoundaries"
+// ret = dateBoundaries.friday()
+// from(bucket: "example-bucket")
+//     |> range(start: ret.start, stop: ret.stop)
+// ```
+//
+// This will return all records from Friday this week
+//
+// ## Metadata
+// tags: date/time
+//
+saturday = () => {
+    return _day_finder(td: date.Saturday, func: _day_formatter)
+}
+
+// sunday will return starting and stopping point of the Sunday this or last week
+// ## Parameters
+//
+//
+// ## Examples
+//
+// ### Find the Sunday of that week
+//
+// ```no_run
+// import "dateBoundaries"
+// option location = timezone.fixed(offset: -8h)
+// option now = () => 2021-12-30T00:40:44Z
+// dateBoundaries.sunday()
+//
+// Returns {start: 2021-12-26T08:00:00Z, stop:2021-12-27T08:00:00Z }
+//
+//
+//
+//
+// ### Filter by the records taken in a day
+// ```no_run
+// import "dateBoundaries"
+// ret = sunday()
+// from(bucket: "example-bucket")
+//     |> range(start: ret.start, stop: ret.stop)
+// ```
+//
+// This will return all records from Sunday this week
+//
+// ## Metadata
+// tags: date/time
+//
+sunday = () => {
+    return _day_finder(td: date.Sunday, func: _day_formatter)
+}
+
+// month will return the timestamps of the start and end of the month in a record
+// ## Parameters
+//
+//
+// ## Examples
+//
+// ### Find the timestamps for the start and stopping point of that month
+//
+// ```no_run
+// import "dateBoundaries"
+// option now = () => 2022-05-10T10:10:00Z
+// dateBoundaries.month()
+// // Returns {start:2022-05-01T00:00:00.000000000Z, stop:2022-06-01T00:00:00.000000000Z}
+// ```
+// ### Query data from this month
+// ```no_run
+// import "dateBoundaries"
+// ret = month()
+// from(bucket: "example-bucket")
+//     |> range(start: ret.start, stop: ret.stop)
+// ```
+//
+// This will return all records from this month
+//
+// ## Metadata
+// tags: date/time
+//
+month = () => {
+    s = date.truncate(t: today(), unit: 1mo)
+    start = _timezone_convert(s: s)
+
+    return {start: start, stop: date.add(d: 1mo, to: start)}
+}
+
+// week will return the timestamps of the week start of a given timestamp truncated to the day and the ending point of that week
+// ## Parameters
+//
+// - start_sunday: Boolean to represent if the month starts on a Sunday or Monday (defaults on Monday)
+//
+// ## Examples
+//
+// ### Find the timestamps for the start and end of that week with the week starting on Monday
+//
+// ```no_run
+//
+// import "dateBoundaries"
+// option now = () => 2022-05-10T00:00:00.000010000Z
+// d = dateBoundaries.week()
+//
+// // Returns {start: 2022-05-09T00:00:00.000000000Z, stop: 2022-05-16T00:00:00.000000000Z}
+// ```
+//
+// ### Find the timestamps for the start and end of that week with the week starting on Sunday
+//
+// ```no_run
+//
+// import "dateBoundaries"
+//  option now = () => 2022-05-10T10:10:00Z
+// d = dateBoundaries.week(start_sunday:true)
+//
+// // Returns {start: 2022-05-08T00:00:00.000000000Z, stop: 2022-05-14T00:00:00.000000000Z}
+// ```
```

Comment:
~~~suggestion
// ### Return start and stop timestamps of the current week starting on Sunday
//
// ```no_run
// import "dateBoundaries"
//
// option now = () => 2022-05-10T10:10:00Z
// 
// dateBoundaries.week(start_sunday:true)
// // Returns {start: 2022-05-08T00:00:00.000000000Z, stop: 2022-05-14T00:00:00.000000000Z}
// ```
~~~